### PR TITLE
Hotfix/portal memory leak

### PR DESCRIPTION
**Pull request by arthurd-micro**
## Steps to reproduce

- Navigate to https://react.fluentui.dev/
- Select Components >> Tooltip on the left sidebar
- Select Custom Mount on the right sidebar
- Open Edge Dev Tools and click on Detached Elements tab
- Click on Collect garbage then click on Get Detached Elements
- Verify there is no detached elements at this point
- Hover over the Custom Mount button in the example
- Click on Collect garbage then click on Get Detached Elements
- Verify there are detached elements .

## Previous Behavior

While using a FluentUI Portal with a custom mount node, a reference for a span node is attached to the custom mount node and never removed. If the portal is removed from the DOM, the reference to the span is kept in memory leading to a Detached Element leak.

The image attached below shows a leak while using the Tooltip component, which uses Portal under the hood. This is the Detached Elements tab from Edge dev tools

detached_elements_tab_portal_fluentui

Below is a screenshot from Memory Tab from Edge dev tools. Tt's possible to see that `parent` is holding the leak. 

memory_tab_portal_fluentui

## New Behavior
After closing the tooltip, no tooltip div element is kept in memory element.

## Related Issue(s)

* Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                 Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `80.888 kB`<br />`25.842 kB` |  `80.959 kB`<br />`25.858 kB` | `71 B` <br />`16 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                                                                    |  `78.742 kB`<br />`25.974 kB` |  `78.813 kB`<br />`25.994 kB` | `71 B` <br />`20 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `79.674 kB`<br />`25.637 kB` |  `79.745 kB`<br />`25.657 kB` | `71 B` <br />`20 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `187.539 kB`<br />`52.638 kB` |  `187.61 kB`<br />`52.658 kB` | `71 B` <br />`20 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        |  `118.422 kB`<br />`35.99 kB` | `118.493 kB`<br />`36.011 kB` | `71 B` <br />`21 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `117.988 kB`<br />`36.475 kB` | `118.059 kB`<br />`36.497 kB` | `71 B` <br />`22 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `121.057 kB`<br />`36.981 kB` | `121.128 kB`<br />`37.003 kB` | `71 B` <br />`22 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 |  `105.54 kB`<br />`32.507 kB` | `105.611 kB`<br />`32.528 kB` | `71 B` <br />`21 B`  |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `11.086 kB`<br />`4.093 kB` |   `11.157 kB`<br />`4.114 kB` | `71 B` <br />`21 B`  |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |   `43.49 kB`<br />`15.213 kB` |  `43.561 kB`<br />`15.233 kB` | `71 B` <br />`20 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |          Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                          |  `85.912 kB`<br />`21.686 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                       |  `52.812 kB`<br />`14.689 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                             |    `15.42 kB`<br />`6.167 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                     |   `69.03 kB`<br />`18.984 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> |  `59.888 kB`<br />`16.652 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               |  `34.379 kB`<br />`11.322 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                       |  `83.464 kB`<br />`24.916 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                    |  `58.849 kB`<br />`16.348 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |    `6.069 kB`<br />`2.053 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                    | `129.317 kB`<br />`35.207 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                               |  `42.971 kB`<br />`11.781 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                    | `127.563 kB`<br />`32.864 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                            |      `81 kB`<br />`20.124 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                      |   `80.429 kB`<br />`19.97 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/06505b5c50c6deedcb36b29417cf35b77da85cfc'>06505b5c50c6deedcb36b29417cf35b77da85cfc</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Avatar-mount_base.html">1628</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Avatar-mount.html">1619</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Button-mount_base.html">1161</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Button-mount.html">1152</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProvider-mount_base.html">1978</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProvider-mount.html">2005</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-mount_base.html">773</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-mount.html">787</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">738</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">778</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/InfoButton-mount_base.html">699</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/InfoButton-mount.html">698</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/MakeStyles-mount_base.html">2380</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/MakeStyles-mount.html">2319</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Persona-mount_base.html">3716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/Persona-mount.html">3622</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/SpinButton-mount_base.html">3129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26435/perf-test-react-components/SpinButton-mount.html">3082</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-portal/src/components/Portal/usePortal.ts [hunk 1]
```diff
@@ -26,6 +26,9 @@ export const usePortal_unstable = (props: PortalProps): PortalState => {
     if (state.virtualParentRootRef.current && state.mountNode) {
       setVirtualParent(state.mountNode, state.virtualParentRootRef.current);
     }
+    return () => {
+      if (state.mountNode) {cleanVirtualParent(state.mountNode);}
```
Comment (ling1726) [inline packages/react-components/react-portal/src/components/Portal/usePortal.ts:30]:
```suggestion
      if (state.mountNode) {setVirtualParent(undefined);}
```
This should also work

Comment (JustSlone) [reply packages/react-components/react-portal/src/components/Portal/usePortal.ts:30]: +1 let's take this approach to save on bundle size

Comment (ling1726) [reply packages/react-components/react-portal/src/components/Portal/usePortal.ts:30]: there's also no need for this to be conditional either AFAIK, you could just clear the previous virtual parent after every effect run

Comment (arthurd-micro) [reply packages/react-components/react-portal/src/components/Portal/usePortal.ts]: @ling1726 Typescript complains if I do not check first. The types don't match.  `state.mountNode` can be null and `setVirtualParent` does not accept a possible null value as first parameter

**Review by ling1726 — APPROVED** [review]

**Pull request merged**